src/fhirtypes/StructureDefinition.ts

```diff
@@ -82,6 +83,29 @@ export class StructureDefinition {
    */
   private _sdStructureDefinition: StructureDefinition;
 
+  private _validationErrors: ValidationError[];
+
+  get validationErrors(): ValidationError[] {
+    return this._validationErrors;
+  }
+
+  valid(): boolean {
+    this._validationErrors = [];
+    this.elements
+      ?.filter(e => !e.valid())
+      .forEach(e => {
+        e.validationErrors?.forEach(err => {
+          this._validationErrors.push(
+            new ValidationError(
+              err.issue,
+              [`StructureDefinition.snapshot.where(id = '${e.id}').first()`, err.fhirPath].join('.')
```

Comment:
Fancy.  You've created a real FHIRPath expression that resolves to the element of interest in the SD. While I find that neat, I am also a total nerd.  I wonder if our average user will find this useful or if they'll find it cryptic? 

Another approach might be to convert the id to a FSH path -- which I think could be done fairly easily by using the `diffId()` and then replacing `:sliceName` with `[sliceName]`.  Then you could do `^${err.fhirPath}` too -- so it looks like a full FSH path...

It's just an idea.  Again, my main concern is that many casual profilers may not be very familiar with FHIRPath (even if they _should_ be).  What do you think?

Reply:
Yeah FSH path makes sense, I'll make the change.